code/modules/mob/mob_movement.dm

```diff
@@ -328,7 +328,7 @@
 			var/atom/movable/AM = A
 			if(AM == buckled) //Kind of unnecessary but let's just be sure
 				continue
-			if(AM.density)
+			if(!AM.CanPass(src))

```

Comment:
Won't this mean that monkey can't use tables to move in zero G? And small animals won't be able to use grille.
We could try `if(!AM.CanPass(src) || AM.density)`